### ISSUE TITLE
lease time not updated

### ISSUE THREAD

**Issue by axet** [miniupnp/miniupnp#199]
Hello!

I setup home openwrt router with miniupnpd, and debuging how external port assigned. Not sure who to blame yet, but seems like miniupnp does not update lease time when I send repeated requests 'AddPortMapping'

To get port I have to provide port information and lease time. For example when I ask to keep port for 5 minutes, I keep in mind I will request port again after 1 minute to keep port assigned to me without interruptions.

Using cat /tmp/upnp.leases I can check how long leases stands and find out lease time not been updated after I send 'AddPortMapping' request.

I'm using https://github.com/syncthing/syncthing/blob/master/lib/upnp library, which should be ok.

**Comment by miniupnp** [comment]
please provide needed information for debugging : 
version of miniupnpd
compile configuration
log files

**Comment by axet** [comment]
https://github.com/openwrt/packages/blob/master/net/miniupnpc/Makefile

miniupnpc version 1.9

when I start my router miniupnpc service will create /tmp/upnp.leases

```
root@OpenWrt:/tmp# cat upnp.leases 
TCP:35670:192.168.54.225:22000:1466783211:syncthing-35670
TCP:33132:192.168.54.142:22000:1466784497:syncthing-33132
TCP:62536:192.168.54.216:22000:1466786358:NAT-PMP 62536 tcp
TCP:38137:192.168.54.216:22000:1466786358:syncthing-38137
root@OpenWrt:/tmp# 
```

Every time I call AddPortMapping service creates a line, 1466786358 - timestamp when lease will be released. I expect when I call AppPortMapping twice, lease time will be updated, but it does't. And lease released at specified time any one take take control for released lease port.

**Comment by miniupnp** [comment]
lease file is created by miniupnpd not miniupnpc.
a bug was fixed in commit 430dcc7b22d593295150210832d29a3912a509
please update to miniupnpd 2.0

**Comment by axet** [comment]
Thank you very much! I'll tell openwrt community about patch!
